### PR TITLE
[#7504] Extract Priority in prependValidator fails on HHVM

### PR DESCRIPTION
**Pull request by rms230**
https://github.com/zendframework/zf2/issues/7504

Pull request for issue: Extract Priority in prependValidator fails on HHVM [#7504]

### CODE REVIEW

**Comment by gianarb** [comment]
Hi! Thanks for this PR! 
Can you try to add a unit test with this implementation? 

Thanks

**Comment by rms230** [comment]
Sure, I'm a little busy at the moment, but will try have them done by tomorrow.

File: src/ValidatorChain.php [hunk 1]
```diff
@@ -156,17 +156,16 @@ public function prependValidator(ValidatorInterface $validator, $breakChainOnFai
...
-            $extractedNode = $queue->extract();
-            $priority = $extractedNode[0] + 1;
+            $extractedNodes = $this->validators->toArray(\Zend\Stdlib\PriorityQueue::EXTR_PRIORITY);
+            rsort($extractedNodes, SORT_NUMERIC);
+            $priority = $extractedNodes[0] + 1;
         }
 
         $this->validators->insert(
-            [
+            array(
```
Comment (Maks3w) [inline src/ValidatorChain.php]: Please undo this change

**Comment by Maks3w** [comment]
There is no need for tests. 3 errors has been fixed in HHVM build

> master: https://travis-ci.org/zendframework/zend-validator/jobs/65683368#L317
> Tests: 1576, Assertions: 3180, Errors: 6, Failures: 3, Incomplete: 5.

> PR: Tests: 1576, Assertions: 3186, Errors: 3, Failures: 3, Incomplete: 5.

File: src/ValidatorChain.php [hunk 2]
```diff
@@ -156,17 +156,16 @@ public function prependValidator(ValidatorInterface $validator, $breakChainOnFai
         $priority = self::DEFAULT_PRIORITY;
 
         if (!$this->validators->isEmpty()) {
-            $queue = $this->validators->getIterator();
-            $queue->setExtractFlags(PriorityQueue::EXTR_PRIORITY);
-            $extractedNode = $queue->extract();
-            $priority = $extractedNode[0] + 1;
+            $extractedNodes = $this->validators->toArray(\Zend\Stdlib\PriorityQueue::EXTR_PRIORITY);
```
Comment (Maks3w) [inline src/ValidatorChain.php]: Please replace `\Zend\Stdlib\PriorityQueue` with `PriorityQueue`
